File: api/schemas/ingredient.py
from sqlmodel import SQLModel


class IngredientDetail(SQLModel):
    id: int
    name: str
    amount: float
    units: str
    details: str | None

    recipe_id: int


class IngredientCreate(SQLModel):
    name: str
    amount: float
    units: str
    details: str | None

    recipe_id: int


class IngredientUpdate(SQLModel):
    name: str | None
    amount: float | None
    units: str | None
    details: str | None

File: src/types/Ingredient.ts
import type { RecipeSlim } from "./Recipe";

export interface IngredientSlim {
  id: number;
  name: string;
  amount: number;
  units: string;
  details?: string;
}

export interface IngredientDetail extends IngredientSlim {
  recipe: RecipeSlim;
}

export interface IngredientCreate {
  name: string;
  amount: number;
  units: string;
  details?: string;
  recipe_id: number;
}
